code/options/Option.cpp

```diff
@@ -0,0 +1,138 @@
+//
+//
+
+#include "Option.h"
```

Comment:
I recommend using lowercase file names.  That was one of the first changes made to the codebase back in 2002.

Reply:
I named the file after the class that is defined within it (although `OptionBase` is also in there so I might move that to a separate file) and that starts with an upper case letter. I see no reason to use a different casing for the file and the class name.

Reply:
It's to make life easier for the Linux folks.  All file names were changed from mixed-case to lowercase back in 2002, and they have been that way ever since.  See for example:
https://www.hard-light.net/forums/index.php?topic=8991.0
https://www.hard-light.net/forums/index.php?topic=23459.msg466635#msg466635

Reply:
All the reasons that are described in those threads is about people accidentally checking in changes that break compilation on Linux because they can't test their changes on that platform.

We do not have that problem anymore since we have our CI setup since every pull request is validated before being merged.

Reply:
All right, fair enough.

Reply:
Even if we can catch that issue now, after having gone through the trouble of standardizing the codebase on lowercase filenames, doesn't keeping that make more sense than just deciding we don't need to anymore?  Personally I hate mixed case projects, in terminal on macs and Linux they tend to show capital filenames first, then lowercase, because they use case sensitive sorting.  It causes me to sometimes take longer to find a file in the list than it should have, not to mention having to remember the case as well when typing out filenames, etc.  I feel like the work that was should not be abandoned just because one particular problem of mixed-case files may no longer be an issue.  Surely we have other classes where the class is capitalized and the containing file is not?

Reply:
I'm sure the standardization was necessary when it was done due to the lacking tooling infrastructure but I do not think the same reasons still apply. Using lower case file names simply because that makes it harder to introduce compile breaks on Linux is not a good reason for using lower case names.

I prefer having file names that reflect the name of the class that is being defined in them which is not possible to do completely if file names always have to be lower case.

Reply:
I didn't mention anything about breaking compilation.